site/src/components/roadmap/Roadmap.tsx

```diff
@@ -0,0 +1,220 @@
+        setJsonData(jsonString);
+
+        const items =
+          responseData?.data?.organization?.repository?.projectV2?.items?.nodes;
+
+        //creates an array of objects with data from github
+        const extractedData: RoadmapDataItem[] = items?.map((item: any) => {
+          const fieldValueNodes = item?.fieldValues?.nodes;
+          const text = getFieldValueByName(fieldValueNodes, "Title");
+          const startDate = getFieldValueByName(fieldValueNodes, "Start Date");
+          const targetDate = getFieldValueByName(
+            fieldValueNodes,
+            "Target Date"
+          );
+          const status = getFieldValueByName(fieldValueNodes, "Status");
+          const issueUrl = item?.content?.url;
+
+          return { text, startDate, targetDate, status, issueUrl };
+        });
+
+        //sets the data to retrieved data or null
+        setRoadmapData(extractedData || []);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const getFieldValueByName = (fieldValueNodes: any[], fieldName: string) => {
+    const fieldValueNode = fieldValueNodes?.find(
+      (node: any) => node?.field?.name === fieldName
+    );
+    return (
+      fieldValueNode?.text || fieldValueNode?.date || fieldValueNode?.name || ""
+    );
+  };
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.content}>
+        <Input
+          placeholder="Search"
+          value={searchQuery}
+          onChange={(event) =>
+            setSearchQuery((event.target as HTMLInputElement).value)
+          }
+          className={styles.bar}
+          startAdornment={<SearchIcon />}
+        />
+
+        {roadmapData !== null && roadmapData.length > 0 ? (
+          <CardView
+            sortedRoadmapData={sortedRoadmapData}
+            searchQuery={searchQuery}
+          />
+        ) : (
+          <Banner status="info">
+            <BannerContent>No data available</BannerContent>
+          </Banner>
+        )}
+      </div>
+    </div>
+  );
+};
+
+export const CardView = ({ sortedRoadmapData, searchQuery }: CardViewProps) => {
+  return (
+    <GridLayout className={styles.cardContainer} columns={3}>
+      <div className={styles.column}>
+        <Heading2 className={styles.heading}>
+          Future <ProgressPendingIcon className={styles.icon} size={1.4} />
+        </Heading2>
+        {sortedRoadmapData
+          .filter((item) => {
+            const startDate = item.startDate ? new Date(item.startDate) : null;
+            const today = new Date();
+            const isFutureItem = !startDate || startDate > today;
+            const matchesSearchQuery =
+              searchQuery === "" ||
+              item.text.toLowerCase().includes(searchQuery.toLowerCase());
+            return isFutureItem && matchesSearchQuery;
+          })
+          .map((item) => {
+            const formattedDate = formatDate(new Date(item.targetDate));
```

Comment:
Looks like each column creates duplicated/boilerplate code
this is probably a refactor opportunity